lib/src/json_schema/validator.dart

```diff
@@ -890,6 +936,14 @@ class Validator {
     }
   }
 
+  _setDynamicParent(JsonSchema child, JsonSchema dynamicParent) {
+    _dynamicParents[child] = dynamicParent;
+  }
```

Comment:
I wonder - suppose `_dynamicParents[child]` already has something set when this is called. Should we return it from this function and then restore it when we're done rather than always removing?

Reply:
We could turn this into a list (stack) and push/pop from that. It would be safer.

Reply:
Sure, although it seems that the calling code that pushes is always the code that pops. So we could forego the list and just use the call stack for the same effect.